UPGRADE-4.0.md

```diff
@@ -160,6 +160,11 @@ HttpKernel
 
  * The `DataCollector::varToString()` method has been removed in favor of `cloneVar()`.
 
+ * The `X-Status-Code` header method of setting a custom status code in the response
```

Comment:
We need a similar message for the `UPGRADE-3.3.md` file which mentions the deprecation.